### ISSUE TITLE
Running make install fails.

### ISSUE THREAD

**Issue by johnburkhard** [mattgodbolt/xania#6]
When running `make` or `make install` getting the following output:
`make: *** No rule to make target 'doorman.o', needed by 'doorman'.  Stop.`

**Comment by snellers** [comment]
@johnburkhard  is that the only output you're seeing, or is it getting past the stage where it creates the "cmake-build-debug" directory?  Are you running Linux?

**Comment by johnburkhard** [comment]
Yes, running Ubuntu 20.04 with gcc 9.3. Here's the full output:

```~/Mud/xania/src main*
❯ make clean
rm -f *.o
rm -f chat/*.o
rm -f xania xania-new doorman
rm: cannot remove 'doorman': Is a directory
make: *** [Makefile:70: clean] Error 1

~/Mud/xania/src main
❯ make install
gcc -c  -Dlibc6 -Wall -g act_comm.c
gcc -c  -Dlibc6 -Wall -g act_info.c
gcc -c  -Dlibc6 -Wall -g act_move.c
gcc -c  -Dlibc6 -Wall -g act_obj.c
gcc -c  -Dlibc6 -Wall -g act_wiz.c
act_wiz.c: In function ‘do_mskills’:
act_wiz.c:1373:74: warning: ‘%s’ directive output may be truncated writing up to 4095 bytes into a region of size between 4075 and 4084 [-Wformat-truncation=]
 1373 |       snprintf(skill_list[lev], sizeof(skill_list[lev]), "\n\rLevel %2d: %s",lev,buf);
      |                                                                          ^~      ~~~
act_wiz.c:1373:7: note: ‘snprintf’ output between 13 and 4117 bytes into a destination of size 4096
 1373 |       snprintf(skill_list[lev], sizeof(skill_list[lev]), "\n\rLevel %2d: %s",lev,buf);
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
act_wiz.c: In function ‘do_mspells’:
act_wiz.c:1459:80: warning: ‘%s’ directive output may be truncated writing up to 4095 bytes into a region of size between 4075 and 4084 [-Wformat-truncation=]
 1459 |             snprintf(spell_list[lev], sizeof(spell_list[lev]), "\n\rLevel %2d: %s",lev,buf);
      |                                                                                ^~      ~~~
act_wiz.c:1459:13: note: ‘snprintf’ output between 13 and 4117 bytes into a destination of size 4096
 1459 |             snprintf(spell_list[lev], sizeof(spell_list[lev]), "\n\rLevel %2d: %s",lev,buf);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
gcc -c  -Dlibc6 -Wall -g ban.c
gcc -c  -Dlibc6 -Wall -g buffer.c
gcc -c  -Dlibc6 -Wall -g challeng.c
gcc -c  -Dlibc6 -Wall -g clan.c
gcc -c  -Dlibc6 -Wall -g comm.c
gcc -c  -Dlibc6 -Wall -g const.c
gcc -c  -Dlibc6 -Wall -g db.c
gcc -c  -Dlibc6 -Wall -g db2.c
gcc -c  -Dlibc6 -Wall -g fight.c
gcc -c  -Dlibc6 -Wall -g flags.c
gcc -c  -Dlibc6 -Wall -g handler.c
gcc -c  -Dlibc6 -Wall -g healer.c
gcc -c  -Dlibc6 -Wall -g info.c
gcc -c  -Dlibc6 -Wall -g interp.c
gcc -c  -Dlibc6 -Wall -g lookup.c
gcc -c  -Dlibc6 -Wall -g magic.c
gcc -c  -Dlibc6 -Wall -g magic2.c
gcc -c  -Dlibc6 -Wall -g mob_prog.c
gcc -c  -Dlibc6 -Wall -g mob_commands.c
gcc -c  -Dlibc6 -Wall -g news.c
gcc -c  -Dlibc6 -Wall -g save.c
gcc -c  -Dlibc6 -Wall -g skills.c
skills.c: In function ‘do_spells’:
skills.c:286:80: warning: ‘%s’ directive output may be truncated writing up to 4095 bytes into a region of size between 4075 and 4084 [-Wformat-truncation=]
  286 |             snprintf(spell_list[lev], sizeof(spell_list[lev]), "\n\rLevel %2d: %s",lev,buf);
      |                                                                                ^~      ~~~
skills.c:286:13: note: ‘snprintf’ output between 13 and 4117 bytes into a destination of size 4096
  286 |             snprintf(spell_list[lev], sizeof(spell_list[lev]), "\n\rLevel %2d: %s",lev,buf);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
skills.c: In function ‘do_skills’:
skills.c:348:80: warning: ‘%s’ directive output may be truncated writing up to 4095 bytes into a region of size between 4075 and 4084 [-Wformat-truncation=]
  348 |             snprintf(skill_list[lev], sizeof(skill_list[lev]), "\n\rLevel %2d: %s",lev,buf);
      |                                                                                ^~      ~~~
skills.c:348:13: note: ‘snprintf’ output between 13 and 4117 bytes into a destination of size 4096
  348 |             snprintf(skill_list[lev], sizeof(skill_list[lev]), "\n\rLevel %2d: %s",lev,buf);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
gcc -c  -Dlibc6 -Wall -g special.c
gcc -c  -Dlibc6 -Wall -g tables.c
gcc -c  -Dlibc6 -Wall -g update.c
gcc -c  -Dlibc6 -Wall -g xania.c
gcc -c  -Dlibc6 -Wall -g phil.c
gcc -c  -Dlibc6 -Wall -g note.c
gcc -c  -Dlibc6 -Wall -g trie.c
gcc -c  -Dlibc6 -Wall -g channels.c
gcc -c  -Dlibc6 -Wall -g wiznet.c
g++ -c  -Dlibc6 -Wall -g Wrapper.cpp
gcc -c  -Dlibc6 -Wall -g -o chat/akey.o chat/akey.C
gcc -c  -Dlibc6 -Wall -g -o chat/allkeys.o chat/allkeys.C
gcc -c  -Dlibc6 -Wall -g -o chat/chatmain.o chat/chatmain.C
gcc -c  -Dlibc6 -Wall -g -o chat/eliza.o chat/eliza.C
g++  -g -o xania-new act_comm.o act_info.o act_move.o act_obj.o act_wiz.o ban.o buffer.o challeng.o clan.o comm.o const.o db.o db2.o fight.o flags.o handler.o healer.o info.o interp.o lookup.o magic.o magic2.o mob_prog.o mob_commands.o news.o save.o skills.o special.o tables.o update.o xania.o phil.o note.o trie.o channels.o wiznet.o Wrapper.o chat/akey.o chat/allkeys.o chat/chatmain.o chat/eliza.o -lcrypt -lnsl
Run make install to set up the new binary
Use mudmgr to start the mud
make: *** No rule to make target 'doorman.o', needed by 'doorman'.  Stop.

~/Mud/xania/src main*
❯ ./mudmgr -s 4000

Xania management script executed by "john" on July-19-2020-20:22:13
Nothing will happen if another mudmgr session is already
running Xania. Use the -r option if you need to restart everything.

~/Mud/xania/src main*
❯ Hmmm...doorman is down...
better start it up then...
Hmmm...Xania is down...
better start it up then...
../install/bin/xania does not exist - have you done make install?```

**Comment by snellers** [comment]
Aha,  try running make from the top directory, not from in the src directory.

**Comment by snellers** [comment]
If you run make install,  it'll compile the binaries and put them in the install/bin dir (useful if you want to execute either using gdb).  If you  run make with no target, it'll start doorman & xania both up. At which point you'll be able to telnet to port 9000.  And if you run make stop,  it will shut them both down.   

And if you did want to execute 'xania' using a debugger, due to a quirk of history,  its current working directory must be the 'areas' directory when doing that (this is something the regular startup scripts take care of normally).  We'll probably make that a bit easier at some point.

**Comment by snellers** [comment]
There's a bunch of other things that are not documented yet either, e.g. how to create/cobble together an Immortal character (any character level 92 to 100) for development or experimental purposes. We can look into adding some tips of that sort soon.

**Comment by johnburkhard** [comment]
Getting some warnings during the build process now, which are being treated as errors:

```~/Mud/xania main 21s
❯ make
cd /home/john/Mud/xania/cmake-build-debug && /usr/bin/cmake ..  -DCMAKE_BUILD_TYPE=debug -DCMAKE_INSTALL_PREFIX=/home/john/Mud/xania/install
-- The C compiler identification is GNU 9.3.0
-- The CXX compiler identification is GNU 9.3.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
-- Build files have been written to: /home/john/Mud/xania/cmake-build-debug
/usr/bin/cmake --build /home/john/Mud/xania/cmake-build-debug
make[1]: Entering directory '/home/john/Mud/xania/cmake-build-debug'
make[2]: Entering directory '/home/john/Mud/xania/cmake-build-debug'
make[3]: Entering directory '/home/john/Mud/xania/cmake-build-debug'
Scanning dependencies of target chat
make[3]: Leaving directory '/home/john/Mud/xania/cmake-build-debug'
make[3]: Entering directory '/home/john/Mud/xania/cmake-build-debug'
[  2%] Building CXX object src/chat/CMakeFiles/chat.dir/akey.C.o
[  4%] Building CXX object src/chat/CMakeFiles/chat.dir/allkeys.C.o
[  6%] Building CXX object src/chat/CMakeFiles/chat.dir/chatmain.C.o
[  8%] Building CXX object src/chat/CMakeFiles/chat.dir/eliza.C.o
[ 11%] Linking CXX static library libchat.a
make[3]: Leaving directory '/home/john/Mud/xania/cmake-build-debug'
[ 11%] Built target chat
make[3]: Entering directory '/home/john/Mud/xania/cmake-build-debug'
Scanning dependencies of target xania
make[3]: Leaving directory '/home/john/Mud/xania/cmake-build-debug'
make[3]: Entering directory '/home/john/Mud/xania/cmake-build-debug'
[ 13%] Building C object src/CMakeFiles/xania.dir/act_comm.c.o
[ 15%] Building C object src/CMakeFiles/xania.dir/act_info.c.o
[ 17%] Building C object src/CMakeFiles/xania.dir/act_move.c.o
[ 20%] Building C object src/CMakeFiles/xania.dir/act_obj.c.o
[ 22%] Building C object src/CMakeFiles/xania.dir/act_wiz.c.o
/home/john/Mud/xania/src/act_wiz.c: In function ‘do_mskills’:
/home/john/Mud/xania/src/act_wiz.c:1372:74: error: ‘%s’ directive output may be truncated writing up to 4095 bytes into a region of size between 4075 and 4084 [-Werror=format-truncation=]
 1372 |       snprintf(skill_list[lev], sizeof(skill_list[lev]), "\n\rLevel %2d: %s",lev,buf);
      |                                                                          ^~      ~~~
/home/john/Mud/xania/src/act_wiz.c:1372:7: note: ‘snprintf’ output between 13 and 4117 bytes into a destination of size 4096
 1372 |       snprintf(skill_list[lev], sizeof(skill_list[lev]), "\n\rLevel %2d: %s",lev,buf);
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/john/Mud/xania/src/act_wiz.c: In function ‘do_mspells’:
/home/john/Mud/xania/src/act_wiz.c:1458:80: error: ‘%s’ directive output may be truncated writing up to 4095 bytes into a region of size between 4075 and 4084 [-Werror=format-truncation=]
 1458 |             snprintf(spell_list[lev], sizeof(spell_list[lev]), "\n\rLevel %2d: %s",lev,buf);
      |                                                                                ^~      ~~~
/home/john/Mud/xania/src/act_wiz.c:1458:13: note: ‘snprintf’ output between 13 and 4117 bytes into a destination of size 4096
 1458 |             snprintf(spell_list[lev], sizeof(spell_list[lev]), "\n\rLevel %2d: %s",lev,buf);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[3]: *** [src/CMakeFiles/xania.dir/build.make:115: src/CMakeFiles/xania.dir/act_wiz.c.o] Error 1
make[3]: Leaving directory '/home/john/Mud/xania/cmake-build-debug'
make[2]: *** [CMakeFiles/Makefile2:135: src/CMakeFiles/xania.dir/all] Error 2
make[2]: Leaving directory '/home/john/Mud/xania/cmake-build-debug'
make[1]: *** [Makefile:130: all] Error 2
make[1]: Leaving directory '/home/john/Mud/xania/cmake-build-debug'
make: *** [Makefile:26: build] Error 2```

**Comment by mattgodbolt** [comment]
That ought to be fixed now (I hope!!) We're still in the process of making it work nicely with all the warning on. Please let us know: right now it's building on GCC 7.5.0 for production (old!) but I've had it building on my Ubuntu 20.04 recently.

We should update the CI to build on a few other compilers...I'll open an issue :)

**Comment by mattgodbolt** [comment]
Scratch that! Was able to repro with a release mode build on another machine! Thanks - will fix!

**Comment by mattgodbolt** [comment]
Should be fixed by 5462b00c2e2d3154f41fe22401a5c88cdbe35198

Please let me know if not. All our string handling is dinosaur-age C...
